cpp/test/Slice/errorDetection/OptionalMembers.err

```diff
@@ -1,30 +1,26 @@
 OptionalMembers.ice:33: missing tag
 OptionalMembers.ice:34: missing tag
 OptionalMembers.ice:35: 'abc' is not defined
-OptionalMembers.ice:36: tag is out of range
-OptionalMembers.ice:37: tag is out of range
-OptionalMembers.ice:38: tag is out of range
+OptionalMembers.ice:36: invalid tag '0x80000000': tags must be less than or equal to '2147483647'
+OptionalMembers.ice:37: invalid tag '-0x80000001': tags cannot be negative
+OptionalMembers.ice:38: invalid tag '-1': tags cannot be negative
 OptionalMembers.ice:40: tag for optional data member 'm8' is already in use
-OptionalMembers.ice:41: tag is out of range
-OptionalMembers.ice:41: invalid tag 'C3'
```

Comment:
The old setup also caused us to emit 2 redundant errors in some situations (see how they're both the same line).
Now we only emit 1 error for each bad value.